tests/NewZealand/AnzacDayTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ *  This file is part of the Yasumi package.
+ *
+ *  Copyright (c) 2015 - 2016 AzuyaLabs
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ *  @author Sacha Telgenhof <stelgenhof@gmail.com>
+ */
+
+namespace Yasumi\Tests\NewZealand;
+
+use DateTime;
+use DateTimeZone;
+
+/**
+ * Class for testing ANZAC day in the New Zealand.
+ */
+class AnzacDayTest extends NewZealandBaseTestCase
+{
+    /**
+     * The name of the holiday
+     */
+    const HOLIDAY = 'anzacDay';
+
+    /**
+     * Tests ANZAC Day
+     *
+     * @dataProvider HolidayDataProvider
+     *
+     * @param int $year the year for which the holiday defined in this test needs to be tested
+     * @param DateTime $expected the expected date
+     */
+    public function testHoliday($year, $expected)
+    {
+        $this->assertHoliday(self::REGION, self::HOLIDAY, $year,
+            new DateTime($expected, new DateTimeZone(self::TIMEZONE)));
+    }
+
+    /**
+     * Returns a list of test dates
+     *
+     * @return array list of test dates for the holiday defined in this test
+     */
+    public function HolidayDataProvider()
+    {
+        return [
```

Comment:
Wow! Nice list... Would it be possible to generate the dates rather than hardcoding them? Would be more sustainable I think :)